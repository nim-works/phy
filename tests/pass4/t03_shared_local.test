discard """
  description: "
    Local variables that are not moved from may share their register with
    local variables from other continuations.
  "
"""
(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1) (Continuations
    (Continuation (Params)
      (Locals (Type 0) (Type 0))
      (Continue 1 (List (Move (Local 0))))
    )
    (Continuation (Params (Type 0))
      (Locals (Type 0))
      (Continue 2 (List))
    )
    (Continuation (Params)))
  )
)