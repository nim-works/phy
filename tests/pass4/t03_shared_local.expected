;$sexp
(TypeDefs
  (Int 8)
  (ProcTy (Void))
)
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1)
    (Locals (Type 0) (Type 0))
    (Continuations
      (Continuation (Params)
        (Locals (Local 0) (Local 1))
        (Continue 1)
      )
      (Continuation (Params)
        (Locals (Local 0) (Local 1))
        (Continue 2)
      )
      (Continuation (Params))
    )
  )
)
